app/controllers/security_group_controller.rb

```diff
@@ -191,11 +174,6 @@ def update
   def update_finished
     security_group_id = session[:async][:params][:id]
 
-    if session[:flash_msgs]
```

Comment:
Why remove this?

Are you sure there can be no flash messages in session from the previous request anymore?

Reply:
I tried some scenarios and from my experience this code is unnecessary/dead because flash messages are already in `@flash_array` in this moment. Also `session[:flash_msgs]` here already is `nil`, in every case I tried (issues with editing Security Group, no credentials defined - also for rules, or also no issues). Also check `update` method which foregoes `update_finished`, there's also only `add_flash` which adds flash message only to `@flash_array`, not to the session.

Reply:
Ah, thanks! :)

Makes sense, the only other flashes in `#update` are on errors, and then the task does not get queued :+1: